dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/registry/impl/ServiceRegistryActor.scala

```diff
@@ -19,100 +19,182 @@ import scala.collection.JavaConverters._
 import scala.compat.java8.OptionConverters._
 
 object ServiceRegistryActor {
-  case class Lookup(name: String)
+  case class Lookup(serviceName: String, portName: Option[String])
   case class Remove(name: String)
   case class Register(name: String, service: ServiceRegistryService)
-  case class Route(method: String, path: String)
+  case class Route(method: String, path: String, portName: Option[String])
   case object GetRegisteredServices
   case class RegisteredServices(services: PSequence[RegisteredService])
   sealed trait RouteResult
-  case class Found(address: InetSocketAddress) extends RouteResult
-  case class NotFound(registry: Map[String, ServiceRegistryService]) extends RouteResult
+  case class Found(address: URI) extends RouteResult
+  case class NotFound(registry: Map[String, (URI, ServiceRegistryService)]) extends RouteResult
+
```

Comment:
The protocol to the actor changed quite a bit. The main changes are:

 * `Lookup` now takes an extra optional argument to search by `portName`
 * `Route` (which is the equivalent to `Lookup` used on the service gateway) also needs the extra arg
 * `RouteResult` payloads are richer (`URI` instead of `InetAddress`).

Reply:
Why change `RouteResult` to contain a `URI` instead of `InetAddress`?

As far as I can tell, the only way it is used now is in code like: `new InetSocketAddress(serviceAddress.getHost, serviceAddress.getPort)` (in other words, converted back into an `InetAddress`).